trunk/NDHMS/Routing/Reservoirs/module_levelpool.F

```diff
@@ -0,0 +1,438 @@
+          tmp1 = oc * oa * sqrt(2. * 9.81 * ( (H+dh1/3.) - oe ) )
+          tmp2 = wc * wl * (dh ** (3./2.))
+          discharge = tmp1 + tmp2
+       else if ( (H+dh1/3) .gt. oe ) then     !! only orifice flow,not full
+          discharge = oc * oa * sqrt(2. * 9.81 * ( (H+dh1/3.) - oe ) )
+       else
+          discharge = 0.0
+       endif
+
+       if (sap .gt. 0.0) then
+          dh2 = ((Itdt_3 - discharge)/sap)*dt
+       else
+          dh2 = 0.0
+       endif
+
+       !-- determine Q(H + 2/3 dh2) from elevation-discharge relationship
+       !-- dh3
+       dh = (H + (0.667*dh2)) - we
+       if (dh .gt. maxWeirDepth) then
+          dh = maxWeirDepth
+       endif
+
+       if (dh .gt. 0.0 ) then              !! orifice and overtop discharge
+          tmp1 = oc * oa * sqrt(2. * 9.81 * ( (H+dh2*0.667) - oe ) )
+          tmp2 = wc * wl * (dh ** (3./2.))
+          discharge = tmp1 + tmp2
+       else if ( (H+dh2*0.667) .gt. oe ) then     !! only orifice flow,not full
+          discharge = oc * oa * sqrt(2. * 9.81 * ( (H+dh2*0.667) - oe ) )
+       else
+          discharge = 0.0
+       endif
+
+       if (sap .gt. 0.0) then
+          dh3 = ((Itdt_2_3 - discharge)/sap)*dt
+       else
+          dh3 = 0.0
+       endif
+
+       !-- determine dh and H
+       dh = (dh1/4.) + (0.75*dh3)
+       H = H + dh
+
+       !-- compute final discharge
+       dh = H - we
+       if (dh .gt. maxWeirDepth) then
+          dh = maxWeirDepth
+       endif
+       if (dh .gt. 0.0 ) then              !! orifice and overtop discharge
+          tmp1 = oc * oa * sqrt(2. * 9.81 * ( H - oe ) )
+          tmp2 = wc * wl * (dh ** (3./2.))
+          discharge = tmp1 + tmp2
+       else if ( H .gt. oe ) then     !! only orifice flow,not full
+          discharge = oc * oa * sqrt(2. * 9.81 * ( H - oe ) )
+       else
+          discharge = 0.0
+       endif
+
+       if(H .ge. maxh) then  ! overtop condition
+          discharge = qi1
+          H = maxh
+       endif
+
+       qo1  = discharge  ! return the flow rate from reservoir
+
+    !#ifdef HYDRO_D
+    !#ifndef NCEP_WCOSS
+    !   ! Water balance check
+    !   qdiff_vol = (qi1+ql-qo1)*dt !m3
+    !   hdiff_vol = (H-Htmp)*sap    !m3
+    !22 format(f8.4,2x,f8.4,2x,f8.4,2x,f8.4,2x,f8.4,2x,f6.0,2x,f20.1,2x,f20.1)
+    !   open (unit=67, &
+    !     file='lake_massbalance_out.txt', status='unknown',position='append')
+    !   write(67,22) Htmp, H, qi1, ql, qo1, dt, qdiff_vol, hdiff_vol
+    !   close(67)
+    !#endif
+    !#endif
+
+    23 format('botof H dh orf wr Q',f8.4,2x,f8.4,2x,f8.3,2x,f8.3,2x,f8.2)
+    24 format('ofonl H dh sap Q ',f8.4,2x,f8.4,2x,f8.0,2x,f8.2)
+
+
+    else   ! ELSE for LAKE_OPT....
+    endif  ! ENDIF for LAKE_OPT....
+
+    return
+
+    ! ----------------------------------------------------------------
+    end subroutine LEVELPOOL
+    ! ----------------------------------------------------------------
+
+
```

Comment:
Do we want to keep the following code?

Reply:
I am keeping and uncommenting. It will be used by the tests.

Reply:
Can we document the tests somewhere? Just a paragraph of overview of what they do and what code/makefiles are involved.

Reply:
this can happen later.